File: array/70.py
#n阶楼梯
class Solution(object):
    def climbStairs(self, n):
        if n<3 :
            return n
        prev,cur = 1,2
        for i in range(3,n+1):
            prev,cur = cur,cur+prev
        return cur
        
def main():
    c = Solution()
    print(c.climbStairs(2))
    print(c.climbStairs(3))
    print(c.climbStairs(4))

if __name__=='__main__':
    main()

File: array/70.js
var climbStairs = function(n) {
  if(n<3){
    return n;
  }
  let prev=1,cur=2,temp=0;
  for(let i=3;i<=n;i++){
    temp = cur ; 
    cur +=prev ;
    prev = temp;
  }
  return cur
};

console.log(climbStairs(4))